src/modules/auth/pages/Register.js

```diff
@@ -0,0 +1,106 @@
+import React, { useState, useEffect } from 'react';
+import { Input, Button, Row, Col, Layout, Form, message } from 'antd';
+import { Link } from 'react-router-dom';
+import Cookies from 'js-cookie';
+import AuthAPIClient from '../services/AuthAPIClient';
+import Footer from '../../../core/components/Footer';
+import '../style/auth.scss';
+import logo from '../../../assests/images/logo.jpeg';
+import { checkConfirmPassword } from '../../../core/helper/validator';
+
+function Register(props) {
+  const [username, setUsername] = useState(null);
+  const [password, setPassword] = useState(null);
+  const [confirmPassword, setConfirmPassword] = useState(null);
+  const [isProcessRegister, setIsProcessRegister] = useState(false);
+
+  const validationRules = {
+    username: [{required: true, message: 'Tên người dùng không được bỏ trống'}],
+    password: [
+      { required: true, message: 'Mật khẩu không được bỏ trống' },
+      { min: 6, message: 'Mật khẩu phải chứa ít nhất 6 kí tự' },
+    ],
+    confirmPassword: [
+      ({ getFieldValue }) => ({
+        validator(_, value) {
+          return checkConfirmPassword(value, getFieldValue('password'));
+        },
+      }),
+    ]
+  }
+
+  useEffect(() => {
+    if (Cookies.get('token')) {
+      props.history.push('/');
+    }
+  }, []);
+
+  const onChangeUsername = (e) => setUsername(e.target.value);
+  const onChangePassword = (e) => setPassword(e.target.value);
+  const onChangeConfirmPassword = (e) => setConfirmPassword(e.target.value);
+
+  const register = async () => {
+    try {
+      setIsProcessRegister(true);
+      await AuthAPIClient.register({ username, password });
+      props.history.push('/login');
+    } catch (error) {
+      message.error(error.response.data.message);
```

Comment:
1. Dangerous. This will only work if the error comes from API call.
2. What if we don't have the message? Should we display a default message in that case instead?